core/src/main/java/nl/nn/adapterframework/webcontrol/api/Webservices.java

```diff
@@ -95,11 +96,14 @@ public Response getLogDirectory() throws ApiException {
 
 		List<Map<String, Object>> wsdls = new ArrayList<Map<String, Object>>();
 		for (Adapter adapter : getIbisManager().getRegisteredAdapters()) {
-			Map<String, Object> wsdlMap = new HashMap<String, Object>(2);
+			Map<String, Object> wsdlMap = null;
 			try {
-				Wsdl wsdl = new Wsdl(adapter.getPipeLine());
-				wsdlMap.put("name", wsdl.getName());
-				wsdlMap.put("extension", getWsdlExtension());
+				if(WsdlUtils.canHaveWsdl(adapter)) { // check eligibility
+					wsdlMap = new HashMap<String, Object>(2);
+					Wsdl wsdl = new Wsdl(adapter.getPipeLine());
```

Comment:
I think it is wise to add a TODO comment here, stating that Wsdl generation should be done on demand, not beforehand.